# Perform read-based k-mer based alignment-free analysis on hologenome dataset

## Table of Contents
0. [Input data](#input)
1. [k-mer based alignment-free sample distance calculation](#d2s)
2. [kraken2](#kraken)

## 0. Input data <a name="input"></a>
- Non-coral reads for each hologenome sample
- ## Required softwares 
     - bwa 
     - kraken2
     - bioawk
 
## 1. k-mer based alignment-free sample distance calculation <a name="d2s"></a>
- Map non-coral reads to 6 genomes of Symbiodiniaceae (each representing their genera) and ITS2 from SymPortal database to recover putative Symbiodiniaceae reads
- Genomes we used include
  - _Symbiodinium_: _Symbiodinium microadriaticum_ CCMP2467 (Nand et al., 2021)
  - _Breviolum_: _Breviolum minutum_ Mf1.05b (Shoguchi et al., 2013) 
  - _Cladocopium_: _Cladocopium proliferum_ SCF055-01 (Chen et al., 2022) 
  - _Durusdinium_: _Durusdinium trenchii_ CCMP2556 (Dougan et al ., 2022)
  - _Effrenium_: _Effrenium voratum_ RCC1521 (Shah et al., 2023)
  - _Fugacium_: _Fugacium kawagutii_ CCMP2468 (Li et al., 2020)

```
bwa index Symb_ref_genome.ITS2.fa 
cd noncoral_reads/
for infile in *_1.fq.gz
do
    base=$(basename ${infile} _1.fq.gz)
    bwa mem Symb_ref_genome.ITS2.fa ${infile} ${base}_2.fq.gz -o symb_mapping/${base}.sam    
    samtools view symb_mapping/${base}.sam -F 4 -b | samtools sort > symb_mapping/${base}.bam 
    samtools fasta symb_mapping/${base}.bam > symb_mapping/${base}.fa
done
```

- Quantify metagenomic variation of Symbiodiniaceaen communities across samples using putative extracted Symbiodiniaceaen sequences 
- d2ssect calculates an alignment-free distance between samples based on frequencies of shared kmers

```
cd  symb_mapping/${base}.fa
for f in *.fa ;do
     jellyfish count -m 21 -s 10000000 $f -o ${f%.fa}.jf
done

d2ssect -l *.jf -f *.fa -o symb_community_d2s.txt 
```
- d2ssect generated distance matrix, representing metagenomic variation of Symbiodiniaceaen communities across samples, and this can be analysed using different dimensionality reduction methods such as Non-metric multidimensional scaling (NMDS) and Distance-based redundancy analysis (db-RDA) or tree-building (e.g. hierarchical clustering and Neighbour joining) 
- Used in Figure 1B and Figure 1C
- See https://github.com/bakeronit/acropora_digitifera_wgs for further details 

## 2. kraken2 <a name="kraken"></a>
- kraken2-based community profiling using custom database containing 20 Symbiodiniaceae genomes
- See https://github.com/iracooke/atenuis_wgs_pub for further details 
- These include assemblies of
  - _Symbiodinium microadriaticum_ (Nand et al., 2021), _S. natans_ (Gonzalez-Pech et al., 2021), _S. necroappetens_ (Gonzalez-Pech et al., 2021), _S. pilosum_ (Gonzalez-Pech et al., 2021), _S. tridacnidorum_ (Gonzalez-Pech et al., 2021), _Symbiodinium_ sp. A4 (_S. 'linucheae'_; Gonzalez-Pech et al., 2021)
  - _Breviolum	minutum_ (Shoguchi et al., 2013)
  - _Cladocopium proliferum_ (Chen et al., 2022) , _Cladocopium_ sp. C15 (Robbins et al., 2019), _Cladocopium_ sp. C92 (Shoguchi et al., 2018)
  - _Durusdinium trenchii _(Shoguchi et al., 2020; Dougan et al 2023)
  - _Effrenium voratum_ (Shah et al., 2023)
  - _Fugacium kawagutii_ (Liu et al., 2018)

### Creating database database 
```
# create empty database 
kraken2-build --download-taxonomy --db symb_kraken_db

# Annotate each genome with NCBI taxonomy id 
bioawk -c fastx '{print ">"$name"|kraken:taxid|2499525\n"$seq}' symbgenomes/Breviolum_minutum_v1.1_Chen_et_al_2020.genome.fa.gz > symb_kraken/Breviolum_minutum_v1.1_Chen_et_al_2020.genome.fa.gz
bioawk -c fastx '{print ">"$name"|kraken:taxid|3025019\n"$seq}' symbgenomes/Cladocopium_goreaui_v2_Chen_et_al_2022.genome.fa.gz > symb_kraken/Cladocopium_goreaui_v2_Chen_et_al_2022.genome.fa.gz 
bioawk -c fastx '{print ">"$name"|kraken:taxid|377605\n"$seq}' symbgenomes/Cladocopium_sp_C15_v1_Robbins_et_al_2019.genome.fa.gz > symb_kraken/Cladocopium_sp_C15_v1_Robbins_et_al_2019.genome.fa.gz
bioawk -c fastx '{print ">"$name"|kraken:taxid|2486705\n"$seq}' symbgenomes/Cladocopium_sp_C92_v1.1_Chen_et_al_2020.genome.fa.gz > symb_kraken/Cladocopium_sp_C92_v1.1_Chen_et_al_2020.genome.fa.gz
bioawk -c fastx '{print ">"$name"|kraken:taxid|1381693\n"$seq}' symbgenomes/Durusdinium_trenchii_CCMP2556_v1_Dougan_et_al_2022.genome.fa.gz  > symb_kraken/Durusdinium_trenchii_CCMP2556_v1_Dougan_et_al_2022.genome.fa.gz
bioawk -c fastx '{print ">"$name"|kraken:taxid|1381693\n"$seq}' symbgenomes/Durusdinium_trenchii_NIES2907_v1_Shoguchi_et_al_2021.genome.fa.gz >  symb_kraken/Durusdinium_trenchii_NIES2907_v1_Shoguchi_et_al_2021.genome.fa.gz
bioawk -c fastx '{print ">"$name"|kraken:taxid|1381693\n"$seq}' symbgenomes/Durusdinium_trenchii_SCF082_v1_Dougan_et_al_2022.genome.fa.gz >  symb_kraken/Durusdinium_trenchii_SCF082_v1_Dougan_et_al_2022.genome.fa.gz
bioawk -c fastx '{print ">"$name"|kraken:taxid|2562239\n"$seq}' symbgenomes/Effrenium_voratum_CCMP421_v2_Shah_et_al_XXXX.genome.fa.gz >  symb_kraken/Effrenium_voratum_CCMP421_v2_Shah_et_al_XXXX.genome.fa.gz
bioawk -c fastx '{print ">"$name"|kraken:taxid|2562239\n"$seq}' symbgenomes/Effrenium_voratum_RCC1521_v2_Shah_et_al_XXXX.genome.fa.gz >  symb_kraken/Effrenium_voratum_RCC1521_v2_Shah_et_al_XXXX.genome.fa.gz
bioawk -c fastx '{print ">"$name"|kraken:taxid|2562239\n"$seq}' symbgenomes/Effrenium_voratum_rt383_v2_Shah_et_al_XXXX.genome.fa.gz >  symb_kraken/Effrenium_voratum_rt383_v2_Shah_et_al_XXXX.genome.fa.gz 
bioawk -c fastx '{print ">"$name"|kraken:taxid|2697096\n"$seq}' symbgenomes/Fugacium_kawagutii_Hic_v3_Li_et_al_2020.genome.fa.gz >  symb_kraken/Fugacium_kawagutii_Hic_v3_Li_et_al_2020.genome.fa.gz
bioawk -c fastx '{print ">"$name"|kraken:taxid|154551\n"$seq}' symbgenomes/Symbiodinium_linucheae_CCMP2456_v1_Gonzalez-Pech_et_al_2021.genome.fa.gz >  symb_kraken/Symbiodinium_linucheae_CCMP2456_v1_Gonzalez-Pech_et_al_2021.genome.fa.gz
bioawk -c fastx '{print ">"$name"|kraken:taxid|2951\n"$seq}' symbgenomes/Symbiodinium_microadriaticum_04-503SCI.03_v1_Gonzalez-Pech_et_al_2021.genome.fa.gz >  symb_kraken/Symbiodinium_microadriaticum_04-503SCI.03_v1_Gonzalez-Pech_et_al_2021.genome.fa.gz 
bioawk -c fastx '{print ">"$name"|kraken:taxid|2951\n"$seq}' symbgenomes/Symbiodinium_microadriaticum_CassKB8_v1_Gonzalez-Pech_et_al_2021.genome.fasta.gz >  symb_kraken/Symbiodinium_microadriaticum_CassKB8_v1_Gonzalez-Pech_et_al_2021.genome.fasta.gz
bioawk -c fastx '{print ">"$name"|kraken:taxid|2951\n"$seq}' symbgenomes/Symbiodinium_microadriaticum_CCMP2467_v2_Nand_et_al_2021_HiC_Smic1.1N.genome.fa.gz >  symb_kraken/Symbiodinium_microadriaticum_CCMP2467_v2_Nand_et_al_2021_HiC_Smic1.1N.genome.fa.gz
bioawk -c fastx '{print ">"$name"|kraken:taxid|878477\n"$seq}' symbgenomes/Symbiodinium_natans_CCMP2548_v1_Gonzalez-Pech_et_al_2021.genome.fa.gz >  symb_kraken/Symbiodinium_natans_CCMP2548_v1_Gonzalez-Pech_et_al_2021.genome.fa.gz
bioawk -c fastx '{print ">"$name"|kraken:taxid|1628268\n"$seq}' symbgenomes/Symbiodinium_necroappetens_CCMP2469_v1_Gonzalez-Pech_et_al_2021.genome.fasta.gz  > symb_kraken/Symbiodinium_necroappetens_CCMP2469_v1_Gonzalez-Pech_et_al_2021.genome.fasta.gz
bioawk -c fastx '{print ">"$name"|kraken:taxid|2952\n"$seq}' symbgenomes/Symbiodinium_pilosum_CCMP2461_v1_Gonzalez-Pech_et_al_2021.genome.fasta.gz  > symb_kraken/Symbiodinium_pilosum_CCMP2461_v1_Gonzalez-Pech_et_al_2021.genome.fasta.gz
bioawk -c fastx '{print ">"$name"|kraken:taxid|1602974\n"$seq}' symbgenomes/Symbiodinium_tridacnidorum_CCMP2592_v1_Gonzalez-Pech_et_al_2021.genome.fa.gz  > symb_kraken/Symbiodinium_tridacnidorum_CCMP2592_v1_Gonzalez-Pech_et_al_2021.genome.fa.gz
bioawk -c fastx '{print ">"$name"|kraken:taxid|1602974\n"$seq}' symbgenomes/Symbiodinium_tridacnidorum_Sh18_v1.1_Chen_et_al_2020.genome.fa.gz  > symb_kraken/Symbiodinium_tridacnidorum_Sh18_v1.1_Chen_et_al_2020.genome.fa.gz

# add these genomes into the kraken database 
kraken2-build --add-to-library symb_kraken/Breviolum_minutum_v1.1_Chen_et_al_2020.genome.fa.gz -db symb_kraken_db --threads 48
kraken2-build --add-to-library symb_kraken/Cladocopium_goreaui_v2_Chen_et_al_2022.genome.fa.gz -db symb_kraken_db --threads 48
kraken2-build --add-to-library symb_kraken/Cladocopium_sp_C15_v1_Robbins_et_al_2019.genome.fa.gz -db symb_kraken_db --threads 48
kraken2-build --add-to-library symb_kraken/Cladocopium_sp_C92_v1.1_Chen_et_al_2020.genome.fa.gz -db symb_kraken_db --threads 48
kraken2-build --add-to-library symb_kraken/Durusdinium_trenchii_CCMP2556_v1_Dougan_et_al_2022.genome.fa.gz -db symb_kraken_db --threads 48
kraken2-build --add-to-library symb_kraken/Durusdinium_trenchii_NIES2907_v1_Shoguchi_et_al_2021.genome.fa.gz -db symb_kraken_db --threads 48
kraken2-build --add-to-library symb_kraken/Durusdinium_trenchii_SCF082_v1_Dougan_et_al_2022.genome.fa.gz -db symb_kraken_db --threads 48
kraken2-build --add-to-library symb_kraken/Effrenium_voratum_CCMP421_v2_Shah_et_al_XXXX.genome.fa.gz -db symb_kraken_db --threads 48
kraken2-build --add-to-library symb_kraken/Effrenium_voratum_RCC1521_v2_Shah_et_al_XXXX.genome.fa.gz -db symb_kraken_db --threads 48
kraken2-build --add-to-library symb_kraken/Effrenium_voratum_rt383_v2_Shah_et_al_XXXX.genome.fa.gz -db symb_kraken_db --threads 48
kraken2-build --add-to-library symb_kraken/Fugacium_kawagutii_Hic_v3_Li_et_al_2020.genome.fa.gz -db symb_kraken_db --threads 48
kraken2-build --add-to-library symb_kraken/Symbiodinium_linucheae_CCMP2456_v1_Gonzalez-Pech_et_al_2021.genome.fa.gz -db symb_kraken_db --threads 48
kraken2-build --add-to-library symb_kraken/Symbiodinium_microadriaticum_04-503SCI.03_v1_Gonzalez-Pech_et_al_2021.genome.fa.gz -db symb_kraken_db --threads 48
kraken2-build --add-to-library symb_kraken/Symbiodinium_microadriaticum_CassKB8_v1_Gonzalez-Pech_et_al_2021.genome.fasta.gz -db symb_kraken_db --threads 48
kraken2-build --add-to-library symb_kraken/Symbiodinium_microadriaticum_CCMP2467_v2_Nand_et_al_2021_HiC_Smic1.1N.genome.fa.gz -db symb_kraken_db --threads 48
kraken2-build --add-to-library symb_kraken/Symbiodinium_natans_CCMP2548_v1_Gonzalez-Pech_et_al_2021.genome.fa.gz -db symb_kraken_db --threads 48
kraken2-build --add-to-library symb_kraken/Symbiodinium_necroappetens_CCMP2469_v1_Gonzalez-Pech_et_al_2021.genome.fasta.gz -db symb_kraken_db --threads 48
kraken2-build --add-to-library symb_kraken/Symbiodinium_pilosum_CCMP2461_v1_Gonzalez-Pech_et_al_2021.genome.fasta.gz -db symb_kraken_db --threads 48
kraken2-build --add-to-library symb_kraken/Symbiodinium_tridacnidorum_CCMP2592_v1_Gonzalez-Pech_et_al_2021.genome.fa.gz -db symb_kraken_db --threads 48
kraken2-build --add-to-library symb_kraken/Symbiodinium_tridacnidorum_Sh18_v1.1_Chen_et_al_2020.genome.fa.gz -db symb_kraken_db --threads 48
```

### Using the kraken database 
```
cd noncoral_reads/
for infile in *_1.fq.gz
do
   base=$(basename ${infile} _1.fq.gz)
   kraken2/kraken2 --threads 48 --db symb_kraken_db --report simu_kbaf/kraken_result/${base}.report \
   --use-mpa-style --report-zero-counts --use-names --gzip-compressed \
   --paired ${infile} ${base}_2.fq.gz > kraken_result/${base}.out
done
```





